### ISSUE TITLE
v.2.4.2 panic: InvalidValue: Invalid OpenGL ES version 2.1

### ISSUE THREAD

**Issue by vinser** [fyne-io/fyne#4419]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

After upgrade to v2.4.2 apps panic with: InvalidValue: Invalid OpenGL ES version 2.1
On v2.4.1 they work as expected

### How to reproduce

Run Example code

### Screenshots

```
vinser@n2p-dev:~/Dev/t_242$ go run .
panic: InvalidValue: Invalid OpenGL ES version 2.1

goroutine 1 [running, locked to thread]:
github.com/go-gl/glfw/v3.3/glfw.acceptError({0x4000897ca0, 0x2, 0x4000897cb8?})
        /home/vinser/go/pkg/mod/github.com/go-gl/glfw/v3.3/glfw@v0.0.0-20221017161538-93cebf72946b/error.go:174 +0x1e8
github.com/go-gl/glfw/v3.3/glfw.CreateWindow(0x4000412000?, 0x400011ec00?, {0xaba182?, 0x95ba38?}, 0x0, 0x0)
        /home/vinser/go/pkg/mod/github.com/go-gl/glfw/v3.3/glfw@v0.0.0-20221017161538-93cebf72946b/window.go:350 +0x144
fyne.io/fyne/v2/internal/driver/glfw.(*window).create.func1()
        /home/vinser/go/pkg/mod/fyne.io/fyne/v2@v2.4.2/internal/driver/glfw/window_desktop.go:737 +0x120
fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).runGL(0x400031ff10)
        /home/vinser/go/pkg/mod/fyne.io/fyne/v2@v2.4.2/internal/driver/glfw/loop.go:136 +0x148
fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).Run(0x400031ff10)
        /home/vinser/go/pkg/mod/fyne.io/fyne/v2@v2.4.2/internal/driver/glfw/driver.go:168 +0x78
fyne.io/fyne/v2/internal/driver/glfw.(*window).ShowAndRun(0x4000412000)
        /home/vinser/go/pkg/mod/fyne.io/fyne/v2@v2.4.2/internal/driver/glfw/window.go:231 +0x2c
main.main()
        /home/vinser/Dev/t_242/main.go:21 +0x210
exit status 2
```

### Example code

```go
package main

import (
	"fyne.io/fyne/v2"
	"fyne.io/fyne/v2/app"
	"fyne.io/fyne/v2/container"
	"fyne.io/fyne/v2/widget"
)

func main() {
	a := app.New()
	w := a.NewWindow("Run 2.4.2")
	w.Resize(fyne.NewSize(400, 100))
	w.CenterOnScreen()

	b := widget.NewButton("Click me!", func() {
		w.SetTitle("Button clicked!")
	})

	w.SetContent(container.NewVBox(b))
	w.ShowAndRun()
}
```

### Fyne version

2.4.2

### Go compiler version

go version go1.21.4 linux/arm64

### Operating system and version

Ubuntu mate 22.04

**Comment by Jacalz** [comment]
OpenGL version problems are very much tied to the hardware (GPU specifically) that you are running it on. Would you mind clarifying what kind of CPU (in the case of integrated graphics) or what kind of GPU you have?

**Comment by andydotxyz** [comment]
The OpenGL ES is not normally picked up by the desktop driver - so the compile command may be important too...

**Comment by Jacalz** [comment]
@andydotxyz The error output show just a `go run .` command already.

**Comment by andydotxyz** [comment]
I have pushed a simple downgrade commit for the desktop ES code - if you could test version `c1a271e9e` then perhaps the issue is resolved?

**Comment by vinser** [comment]
@andydotxyz Great!  `go: downgraded fyne.io/fyne/v2 v2.4.2 => v2.4.2-rc1.0.20231124101500-c1a271e9e294` really did the job. Was tested with above code sample and another small app where I catch the bug.
Thanks

**Comment by aykevl** [comment]
Got the same issue here at version `fyne.io/fyne/v2 v2.4.2` and reverting to 2.4.1 also fixes it for me.

The c1a271e9e version also works for me:

```
$ go get fyne.io/fyne/v2@c1a271e9e
go: downloading fyne.io/fyne/v2 v2.4.2-rc1.0.20231124101500-c1a271e9e294
go: upgraded fyne.io/fyne/v2 v2.4.1 => v2.4.2-rc1.0.20231124101500-c1a271e9e294
```

I am running Fedora Asahi, so I'm using an Apple GPU (M1 Pro specifically), which is currently at OpenGL version 3.3. I can share more details if needed.

**Comment by andydotxyz** [comment]
Thanks for confirming. I will look into maybe a 2.4.3 sooner than planned - there is another annoying bug that I think we should address.

**Comment by lordofscripts** [comment]
Took my system quite a while to download and compile Fyne v2.4.2 (I had v2.3.5 or so) and I got the same error as the OP when I tried to run my GoFyne application.

My system is a Raspberry Pi 2B (linux/arm 32-bit - Debian/Bullseye).

**Comment by andydotxyz** [comment]
The fix is not released yet. You can test it by using the @develop version for your Fyne dependency.

**Comment by lordofscripts** [comment]
I downgraded to v2.4.1 and the error disappeared.

**Comment by andydotxyz** [comment]
Yes it would, the error occurred in v2.4.2 and will be resolved in v2.4.3. But the fix is already on develop as I mentioned above.
